auth/Program.cs

```diff
@@ -0,0 +1,141 @@
+﻿/*
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+// [START all]
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Storage.v1;
+using Google.Apis.Storage.v1.Data;
+using Google.Apis.Services;
+using System;
+using System.Threading.Tasks;
+
+namespace GoogleCloudSamples
+{
+    public class AuthSample
+    {
+        const string usage = @"Usage:AuthSample <bucket_name>";
+        // [START build_service]
+        /// <summary>
+        /// Creates an authorized Cloud Storage client service using Application 
+        /// Default Credentials.
+        /// </summary>
+        /// <returns>an authorized Cloud Storage client.</returns>
+        public StorageService CreateAuthorizedClient()
+        {
+            GoogleCredential credential =
+                GoogleCredential.GetApplicationDefaultAsync().Result;
+            // Inject the Cloud Storage scope if required.
+            if (credential.IsCreateScopedRequired)
+            {
+                credential = credential.CreateScoped(new[]
+                {
+                    StorageService.Scope.DevstorageReadOnly
+                });
+            }
+            return new StorageService(new BaseClientService.Initializer()
+            {
+                HttpClientInitializer = credential,
+                ApplicationName = "DotNet Google Cloud Platform Auth Sample",
+            });
+        }
+        // [END build_service]
+
+        // [START list_storage_bucket_contents]
+        /// <summary>
+        /// List the contents of a Cloud Storage bucket.
+        /// </summary>
+        /// <param name="bucket">the name of the Cloud Storage bucket.</param>
+        ///<returns>a list of the contents of the specified bucket.</returns>
+        public Objects ExecuteApiRequest(

```

Comment:
Can we rename this to something more specific like ListBuckets()?  If you're trying to communicate that this function is the one that actually calls our API, then maybe name it CallListBuckets()?